### PR TITLE
Don't show undefined in error messages when error details incomplete

### LINKED ISSUE #2130: `Undefined` appears in error details when creating from template
Seen as part of #2072, but a separate issue

When the create fails, the first error shown in the UI has a "more details" link. When I click that, I see "undefined undefined". We should make the template more robust to unspecified values

### PR DESCRIPTION
**Pull request by spadgett**
openshift_management_console

Fixes #2130

### CODE REVIEW

**Comment by spadgett** [comment]
@liggitt please review

File: assets/app/scripts/controllers/newfromtemplate.js [hunk 1]
```diff
@@ -181,4 +176,17 @@ angular.module('openshiftConsole')
...
+      if (!failure.data || !failure.data.details) {
+        return null;
+      }
+
+      var details = failure.data.details;
+      if (details.kind) {
+        return (details.id) ? details.kind + " " + details.id : details.kind;
+      }
+
+      return details.id;
```
Comment (liggitt) [inline assets/app/scripts/controllers/newfromtemplate.js]: we're not guaranteed a details id are we?

Comment (spadgett) [inline assets/app/scripts/controllers/newfromtemplate.js]: @liggitt No, but we handle when this method returns null or undefined above (line 118).

**Comment by liggitt** [comment]
[merge]

**Pull request merged**